src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let key = 0; key < actions.length; key++) {
+    const action = actions[key];
+
+    if (action.type === 'addProperties') {
+      Object.assign(state, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (let i = 0; i < action.keysToRemove.length; i++) {
+        if (action.keysToRemove[i] in state) {
+          delete state[action.keysToRemove[i]];
+        }
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const value in state) {
+        delete state[value];
+      }
+    }
```

Comment:
If you have predetermined amount of options it is better to use switch/case structure instead of multiple
if/else blocks